### DISCUSSION TITLE
Selectable clipPath

### DISCUSSION THREAD

**Discussion by Jiiiin** [fabricjs/fabric.js#7849] [Q&A] [answered]
It's very common in Photoshop, I have two Objects in canvas, one is a shape described by path, another is a image. The image is drawn only where it overlaps the shape,everything else is made transparent. I knew clipPath, but it just clip the image , I want the image to be complete and both the shape and the picture to be draggable.

I also tried using GlobalCompositeOperation, but I couldn't find a suitable property to set on Object

![image](https://user-images.githubusercontent.com/22162992/161927509-c2eb2b00-6e9d-4a56-a997-94ced59e4ce0.png)

### The effect i want:
https://user-images.githubusercontent.com/22162992/161950247-1a96d2e6-b7dd-4b4d-8fb0-37081d01e60b.mov

### My project

https://codesandbox.io/s/eloquent-perlman-k2epxe?file=/src/App.tsx

**Comment by ShaMan123** [comment, marked as answer]
clipPath

**Reply by ShaMan123** [reply]
https://codesandbox.io/s/relaxed-goodall-hq1t47?file=/src/App.tsx

**Comment by ShaMan123** [comment]
You should put some effort in your questions if you want anyone to put some effort in the answers

**Reply by Jiiiin** [reply]
sry, i try to refine my question，do you have any suggestions to help me？

**Reply by ShaMan123** [reply]
Post what you have tried, the resources on fabric you have exhausted, write about your efforts and failures.
This is a crucial place to dig at http://fabricjs.com/demos/
Create a working reproduction, post a video.
Many things that can be done.
Mainly, it's hard to understand what you mean so put a lot of effort into making things extremely clear.

**Reply by Jiiiin** [reply]
ok i uploaded a video and posted a demo project in my question,thank you very much.

**Comment by ShaMan123** [comment]
https://codesandbox.io/s/relaxed-goodall-hq1t47?file=/src/App.tsx

**Reply by Jiiiin** [reply]
This is what I want, thank you so much

**Comment by ShaMan123** [comment]
I have added this example to http://fabricjs.com/clipping
